### PR TITLE
ClientRequests -> OutgoingHttpDependencies

### PR DESCRIPTION
**Pull request by joshgav**
see https://github.com/Microsoft/ApplicationInsights-node.js/issues/225 for background.

### CODE REVIEW

**Comment by joshgav** [comment]
cc @jasongin

File: package.json [hunk 1]
```diff
@@ -40,9 +40,9 @@
     }
   ],
   "scripts": {
-    "build": "tsc --module commonjs --declaration applicationinsights.ts",
-    "prepublish": "tsc --module commonjs --declaration applicationinsights.ts",
-    "pretest": "find Tests -type f -name \"*.ts\" | xargs tsc --module commonjs",
+    "build": "tsc --module commonjs --target es5 --declaration applicationinsights.ts",
```
Comment (OsvaldoRosado) [inline package.json]: Does this have any compat issues to be aware of in old node?

**Comment by OsvaldoRosado** [comment]
While we're renaming this, why not just HttpDependencies?

Also, for some historical context there's some conversation about this naming at #129 .  Function names were changed to be in line with request/dependency as used in our other SDKs, but these class namings are from the original structure. I think your PR would provide a good cleanup.

**Comment by joshgav** [comment]
@OsvaldoRosado 

> While we're renaming this, why not just HttpDependencies?

Indeed 😄 , I actually had that as an intermediate class at some point along the way, but then realized that incoming HTTP requests will always be Requests in the App Insights sense and didn't want outgoing requests, which will always be RemoteDependencies in the App Insights sense, to be confused. For the same reason I kept the actual class as OutgoingHttpDependencies so that no one makes the mistake of thinking this relates to incoming HTTP requests. I still lean to keeping the longer name cause I think it would be best to be as clear as possible in this area, since we have to reason about it a lot for correlation.

Will check out that other issue too, thanks!

File: package.json (hunk 1 above)
Comment (joshgav) [inline package.json]: According to CI the only problem is with use of Object.keys() deep in a helper method. I'll check that out, but it seems to indicate the rest is okay?

**Comment by joshgav** [comment]
All green 🎉 ! PTAL.

File: AutoCollection/OutgoingHttpDependencies.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                !request['getHeader'](HttpHeaders.RemoteDependency.myIkey)) {
+            request['setHeader'](HttpHeaders.RemoteDependency.myIkey, outgoingHttpDependencyParser.myIkeyHash);
+        }
+
+        // Collect dependency telemetry about the request when it finishes.
+        if (request.on) {
+            request.on('response', (response: http.IncomingMessage) => {
+                outgoingHttpDependencyParser.onResponse(response, properties);
+                // TODO(joshgav): Is it okay to change ClientResponse -> IncomingMessage?
+                var context : { [name: string]: any; } = { "http.RequestOptions": requestOptions, "http.ClientRequest": request, "http.ClientResponse": response };
```
Comment (OsvaldoRosado) [inline AutoCollection/OutgoingHttpDependencies.ts]: Do you know when Node switched from ClientResponse to IncomingMessage? I see ClientResponse isn't documented anymore

Comment (joshgav) [inline AutoCollection/OutgoingHttpDependencies.ts]: not completely sure but it's in the index.d.ts file: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v0/index.d.ts#L450-L453

the `connection` property is also gone, have been updating that around the codebase too

**Comment by OsvaldoRosado** [comment]
Closing as this was addressed in #309

**Pull request closed without merging**